Index: AxisServlet.java
===================================================================
RCS file: /home/globdev/CVS/globus-packages/wsrf/java/core/source/src/org/globus/wsrf/container/AxisServlet.java,v
retrieving revision 1.8.2.3
diff -u -r1.8.2.3 AxisServlet.java
--- AxisServlet.java	26 Jul 2005 22:52:24 -0000	1.8.2.3
+++ AxisServlet.java	20 Jul 2007 17:50:27 -0000
@@ -66,10 +66,15 @@
         ServletConfig servletConfig = getServletConfig();
         ServletContext context = servletConfig.getServletContext();
 
-        // XXX: always overwrite web context
-        File rootFile = new File(context.getRealPath ("/"));
-        String rootContext = rootFile.getName();
+        String rootContext = 
+            servletConfig.getInitParameter(ContainerConfig.WEB_CONTEXT);
+        if (rootContext == null) {
+            // use the directory name as the context
+            File rootFile = new File(context.getRealPath ("/"));
+            rootContext = rootFile.getName();
+        }
             
+        // XXX: always overwrite web context
         config.setOption(ContainerConfig.WEB_CONTEXT, rootContext);
 
         String webInfPath = getWebInfPath();